### ISSUE TITLE
[Bug]: SELECT coalesce(starred, 0) as starred, … Scan error on column index 43, name \"bpm\": converting NULL to int is unsupported

### ISSUE THREAD

**Issue by grinapo** [navidrome/navidrome#2825]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.51.0

### Current Behavior

Doesn't cause too much harm (probably broken media links), but it's quite filling the log at startup (see below) with

`error="sql: Scan error on column index 43, name \"bpm\": converting NULL to int is unsupported"`

Seems like some db upgrade didn't go well. 

By the way I have heaps of warnings, may be related:
```
2024-01-29_10:29:27.56783 time="2024-01-29T11:29:27+01:00" level=warning msg="Error warming cache" error="error cacheing id='al-6b92b65462aebc5839584480e5f1ba89_4cfbae08': could not get a cover art for al-6b92b65462aebc5839584480e5f1ba89_4cfbae08: artwork unavailable"
```

### Expected Behavior

No errors. Also startup time may be longer because of that, can't tell.

### Steps To Reproduce

1. start navidrome
2. watch log :-)

### Environment

```markdown
- OS:Debian/sid
- Browser: none
- Client: none
```

### How Navidrome is installed?

Binary (from downloads page)

### Configuration

```toml
MusicFolder = "/wherever/music/"
Scanner.Extractor = "ffmpeg"
```

### Relevant log output

```shell
2024-01-29_10:29:28.25662 time="2024-01-29T11:29:28+01:00" level=error msg="SQL: `SELECT coalesce(starred, 0) as starred, coalesce(rating, 0) as rating, coalesce(play_count, 0) as play_count, starred_at, play_date, media_file.*, coalesce(position, 0) as bookmark_position FROM
 media_file LEFT JOIN annotation on (annotation.item_id = media_file.id AND annotation.item_type = 'media_file' AND annotation.user_id = 'f484d214-1262-4ee7-a99d-89729735e714') LEFT JOIN bookmark on (bookmark.item_id = media_file.id AND bookmark.item_type = 'media_file' AND b
ookmark.user_id = 'f484d214-1262-4ee7-a99d-89729735e714') WHERE album_id IN ({:p0},{:p1},{:p2},{:p3},{:p4},{:p5},{:p6},{:p7},{:p8},{:p9},{:p10},{:p11},{:p12},{:p13},{:p14},{:p15},{:p16},{:p17},{:p18},{:p19},{:p20},{:p21},{:p22},{:p23},{:p24},{:p25},{:p26},{:p27},{:p28},{:p29}
,{:p30},{:p31},{:p32},{:p33},{:p34},{:p35},{:p36},{:p37},{:p38},{:p39},{:p40},{:p41},{:p42},{:p43},{:p44},{:p45},{:p46},{:p47},{:p48},{:p49},{:p50},{:p51},{:p52},{:p53},{:p54},{:p55},{:p56},{:p57},{:p58},{:p59},{:p60},{:p61},{:p62},{:p63},{:p64},{:p65},{:p66},{:p67},{:p68},{:
p69},{:p70},{:p71},{:p72},{:p73},{:p74},{:p75},{:p76},{:p77},{:p78},{:p79},{:p80},{:p81},{:p82},{:p83},{:p84},{:p85},{:p86},{:p87},{:p88},{:p89},{:p90},{:p91},{:p92},{:p93},{:p94},{:p95},{:p96},{:p97},{:p98},{:p99})`" args="map[p0:6e30c3adc15ed7ddc6459f5d45e3ab9b p1:ea0bfe372
1aa0c943eda2501da0692aa p10:42273b17aec34ce7df87b8217c96ae36 p11:cb6b012a553d2fccb1fff4b4665a7b91 p12:75e49e0c4466cf790fc848a5c2bd3da5 p13:0ba9249e033d8acecabb120093b95c96 p14:13bb710dd1b89a050072243dffdaf353 p15:efdcc85ca460e5aa2499307eb3e0a8e4 p16:9080f5ac597ac5761de8bb7d86
332a91 p17:7f061cfb9ae7b292b597021e06015944 p18:d6f2e67009217f02d8478bb0b35d0c03 p19:5c29b503a0b22bc2268085f1a263057a p2:e5c88187ad2f1b69dd5ff333682d5d5d p20:1bb663f49e454915bca8885713e94011 p21:6228e53fd27779e90ea55dcb91cd68f0 p22:0744264fbe35644656b1d84c68caf7f8 p23:0daf058
aad87f986521348628f6d1814 p24:57e3903f0122a9ff02dc7eef37d5a12e p25:b632860038b7c2215b937f60578b593c p26:f0c60daf646b488e478932f59fe11dbe p27:9e3fe36355bb34508bae2354657ecf8f p28:3ccc2ea6ccf08bc007324338e4993ef2 p29:064a520b6f624814d155621a02357016 p3:92379bbae21165da28114cbe2
8e0d905 p30:7631c05ba32bba3413911d9b6f68a14d p31:eb44d87ebd1775713bc7e20429425e7c p32:f618407b9445a7fdc6f544d8bf432af9 p33:ddb99414dbf48837e8c8a03f0b8efa31 p34:2ba00bc14813c4ec9d4a33e349ce4da8 p35:6234286d747e5b641b44b652bd5a5496 p36:5ad17995b5a23d2b0aad0c7eb9290915 p37:17c71
77b5785cde7fd66ee0a387a9c45 p38:c84526822a974d119f76663d891ea269 p39:57f603cdeda8e3e7725e6df60818d9af p4:7f0f91dc40caf76004e7aa1963ba8293 p40:90fa488daab666f259489e1104b98c0d p41:7bf47b9b9e1fd0500c29d4731da0721e p42:568f2eff25288b696fbac30a460373ee p43:1a90b951878ab98da3bad71
caefc2c35 p44:e853f98165ea4a74b7b212ddb6fae87e p45:ead8e7e0f63d79dc3e2bc573d590f3e8 p46:49f9b70231be86e5991bc01e50dd3a2b p47:e9d1775c2d6db57504395dc4568774c6 p48:0dde964ec4bf86c3e593ec46dbd7b454 p49:64e10046b136b3d43ea33a54f1ca5727 p5:514cdebeee31b2a7d5a590e2e1deb213 p50:3bd7
8bdb73e868e95b2b9a176eb1d387 p51:39576a33921b53bc2d13ea4773e5cd8a p52:21f39332647c8ac2cf3e77a03d9edf6a p53:8b23b5a7714b52a008c43b8c50bf840d p54:6a0cf1c6a540a283a2d0bca1a7c83074 p55:ddc6bc9a9f7b27c47deaa6ab206f43df p56:9d00e2bfdd1b438ed3d12546351935ff p57:a589a450dadabc9d7467e
f19c1b7da8d p58:1e622211baa1f9abcffcc3edad6dab9a p59:8d46ff3d5a67f9facde7eecf0811fc30 p6:120c00adc7b9828c59b085d396f81ba2 p60:c3fe1b21bd42d773e7a51a879b0c78cd p61:a111adca75b8e020daeda1f3062f4d02 p62:863167eedb1ac77b3610c960a0a4d124 p63:3497a940cacb4c136debd40c24b644de p64:e0
b22069134d918128492521e1223b69 p65:5e61437db0c234fbfc982e034de1d3e1 p66:34c196b08c1f819f1a9a0a7456eb554e p67:ac38b2acd6c2df4f0906c42c73a78943 p68:117d4fee80517fa7f39c19c4cd650f4f p69:d942e856c48c1bae3111392115aac280 p7:3b226daaf3bad7fc0f134d9476f8cc00 p70:45d215b768d3e65380e9
d6bbad110f04 p71:211030e37584290352f45f998095aaad p72:c00cf5cc9148942f5aaab8bfd2ff52f2 p73:acfd915dae31dc615c65a4160f0bc5ec p74:28104f4ddbb4d5407db8e84adc133a2f p75:377d91bbbd2e6719375e82295c4de6d8 p76:10fe6f0f11efce23dee55f3dd2cb3410 p77:dda0bc22b0007b1b982484f2adbdc3e8 p78:
d8cbbda0344b472fd77a3ff7b66fc077 p79:3abf2a9cb32736f0382205a5b659a1ed p8:ffaac4b71601564586923421bd2d8c99 p80:e9edf4203344507dff8b7890682a2e27 p81:3851540e1ce957fc830614cbb55694a7 p82:07231436b662b27274d3f4954e122083 p83:1a8e0385cf45d77073bc2c15236d1aae p84:fc87e02830010fee7b
fb95064afe6231 p85:cc7c986a660ad5904e22f0fee728b2bc p86:30c7fdbce8da90dad4412730550adff9 p87:63298ecd91cddfbb9e0d3a0f5c875e54 p88:4af57790faaa3d0e17c22234c073c9d2 p89:ef969a69d9aa0226cffb772a809516f9 p9:6b92b65462aebc5839584480e5f1ba89 p90:fdee3ea6728ed18bc8051fc74e4df8f0 p91
:b9776fd5554417393b7b5b6aa54ebc38 p92:746903f6f2e2978edad0e708ac7f5d8e p93:79b24abe9859a60df2c83d5729d24456 p94:13c0431642a05253ae31f7da38ccfd76 p95:83b5745d2a5847b7f89a9efcf89e8584 p96:9ac931f7a95c89a6ff5b6fb81554076a p97:e8945ad8b6b5500a831c699bf2036275 p98:57a66d7ed1189dfc
bb3eef94c5d31335 p99:9d95822a79d52c21ca848835210bb9cf]" elapsedTime=22.9ms error="sql: Scan error on column index 43, name \"bpm\": converting NULL to int is unsupported" requestId=narya.grin.hu/acSFdWLnFm-000024 rowsAffected=-1 username=grin
```

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by grinapo** [comment]
(Similar but got this through the API requesting song sorting:
```json
{"error":"sql: Scan error on column index 47, name \"rg_album_gain\": converting NULL to float64 is unsupported"}null
```
which is an invalid json as well.)

**Comment by deluan** [comment]
We have a bug fix release coming out this week that will hopefully fix all these `converting NULL` errors.

@grinapo , which endpoint was this?

**Comment by grinapo** [comment]
Since it turned out that these break the lookups completely here are the quick fix for the readers:
```
# stop navidrome
sqlite3 navidrome.db
```

```sql
delete from  media_file  where bpm is null or rg_album_gain is null or rg_track_gain is null;
```

and do a full rescan afterwards.

**Comment by deluan** [comment]
This is related to #2806 and will be fixed by #2840. Once the PR is merged we will release a new bug-fix version.
